### ISSUE TITLE
cannot open PDF-file over Samba

### ISSUE THREAD

**Issue by edmundlaugasson** [MeanEYE/Sunflower#250]
When file is located at Samba share (e.g. smb://server/share/file.pdf) then PDF-viewer opens it but gives an error message "Document cannot be opened. The operation is not supported".
When to copy onto local hard drive then file opens without problems in same system.

Used software:
OS: 64-bit Linux Mint 18 MATE
PDF-viewer: XReader 1.0.8 (default PDF-viewer)
Sunflower FM: 0.3 (61)

**Comment by MeanEYE** [comment]
Did you try with different PDF reader? Perhaps yours doesn't support URI
paths.

**Comment by edmundlaugasson** [comment]
Yes, also Atril (default originally in MATE desktop environment) Evince (default originally in GNOME desktop environment) did not work.

**Comment by MeanEYE** [comment]
Okay. So that means we have a bug with URI forming. Can I ask you to grab [latest development](https://github.com/MeanEYE/Sunflower/archive/0.3-gtk2.zip) release and test? You just need to run `Sunflower.py`.

**Comment by edmundlaugasson** [comment]
Tested - same problem.
Some feedback from CLI:
```
$ python Sunflower.py 
Vector smash protection is enabled.
openjdk version "1.8.0_91"
OpenJDK Runtime Environment (build 1.8.0_91-8u91-b14-0ubuntu4~16.04.1-b14)
OpenJDK 64-Bit Server VM (build 25.91-b14, mixed mode)
```

**Comment by MeanEYE** [comment]
Can you please provide the full path you are trying to open?

**Comment by edmundlaugasson** [comment]
Sunflower.py is located at /home/user/Downloads/Sunflower-0.3-gtk2/ folder.
PDF-file is located at smb://server/share$/folder1/folder2/file.pdf

**Comment by MeanEYE** [comment]
Am guessing `$` sign is what is confusing it. Okay, thanks. I think this issue has been fixed in GTK3 version. Can you download [this version](https://github.com/MeanEYE/Sunflower/archive/develop.zip) and check? Thanks!

**Comment by edmundlaugasson** [comment]
Downloaded, tested - still same problem persist.

**Comment by MeanEYE** [comment]
Okay, thank you. So it's a bug which hasn't been addressed yet. I'll look into ti.

**Comment by cmallwitz** [comment]
I would assume this is the same as issue #253 - it is the URL prefix that is causing the issue not the $ character in the path. I tried a local PDF file and added a $ to the file name and parent directory and couldn't find any issue.

Christian

**Comment by defencedog** [comment]
**SOLUTION**
@ Debian x86

The problem will not occur if you browse to this directory & then open files in the mounted SAMBA filesystem

`/run/user/1000/gvfs/smb-share:server=[USER],share=[SAMBA_DIR]`

If you mount SAMBA via nautilus automatically folder for that drive is generated under
`/run/user/1000/gvfs/`

I cannot open any file in a SAMBA mounted drive by navigating to mounted filesystems shown in sunflower

**Comment by MeanEYE** [comment]
This seems to be the problem with URI formatting. Thanks for narrowing it down @defencedog.
